src/include/dawg/root.h

```diff
@@ -5,27 +5,38 @@
  *  Copyright (C) 2010 Reed A. Cartwright, PhD <reed@scit.us>               *
  ****************************************************************************/
 
+#include <algorithm>
+
 #include <dawg/mutt.h>
 #include <dawg/subst.h>
 #include <dawg/rate.h>
-#include <dawg/sequence.h>
+#include <dawg/residue.h>
 
 namespace dawg {
 
 class root_model {
 public:
-	bool create(unsigned int len, const std::string &seq,  const std::vector<double> &rates) {
+	bool create(unsigned int len, sequence &&root_seq) {
```

Comment:
Do not use an rvalue reference in arguments. If you want to move something use copy by value in the arguments so the caller can decide whether they want to move it or not.

Reply:
I should note that after modifying the rvalue reference it won't compile because in `matic.cpp` it won't bind to a temporary value (without the move), but I fixed it by changing the caller to not use a temporary.